sound/soc/sof/ipc4-topology.c

```diff
@@ -1051,15 +1051,19 @@ static int sof_ipc4_init_audio_fmt(struct snd_sof_dev *sdev,
 	}
 
 	/* copy input format */
-	memcpy(&base_config->audio_fmt, &available_fmt->input_pin_fmts[i].audio_fmt,
-	       sizeof(struct sof_ipc4_audio_format));
+	if (available_fmt->input_pin_fmts)
+		memcpy(&base_config->audio_fmt, &available_fmt->input_pin_fmts[i].audio_fmt,
+		       sizeof(struct sof_ipc4_audio_format));
 
 	/* set base_cfg ibs/obs */
-	base_config->ibs = available_fmt->input_pin_fmts[i].buffer_size;
-	base_config->obs = available_fmt->output_pin_fmts[i].buffer_size;
+	if (available_fmt->input_pin_fmts)
+		base_config->ibs = available_fmt->input_pin_fmts[i].buffer_size;
+	if (available_fmt->output_pin_fmts)
+		base_config->obs = available_fmt->output_pin_fmts[i].buffer_size;
```

Comment:
@libinyang, if there is no output at all, then obs might be better to set to 0?

Reply:
> @libinyang, if there is no output at all, then obs might be better to set to 0?

The default value is 0 as we are using kzalloc()

Reply:
right, and what if we have 1 input pin and the i is 2? What if we have 1 output pin and the i is 2?